contrib/storage-kafka/src/main/java/org/apache/drill/exec/store/kafka/KafkaNodeProcessor.java

```diff
@@ -0,0 +1,186 @@
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+class KafkaNodeProcessor extends AbstractExprVisitor<Boolean, LogicalExpression, RuntimeException> {
+
+  private String functionName;
+  private Boolean success;
+  private Long value;
+  private String path;
+
+  public KafkaNodeProcessor(String functionName) {
+    this.functionName = functionName;
+    this.success = false;
+  }
+
+  public static boolean isPushdownFunction(String functionName) {
+    return COMPARE_FUNCTIONS_TRANSPOSE_MAP.keySet().contains(functionName);
+  }
+
+  @Override
+  public Boolean visitUnknown(LogicalExpression e, LogicalExpression valueArg) throws RuntimeException {
+    return false;
+  }
+
+  public static KafkaNodeProcessor process(FunctionCall call) {
+    String functionName = call.getName();
+    LogicalExpression nameArg = call.args.get(0);
+    LogicalExpression valueArg = call.args.size() >= 2? call.args.get(1) : null;
+    KafkaNodeProcessor evaluator = new KafkaNodeProcessor(functionName);
+
+    if (VALUE_EXPRESSION_CLASSES.contains(nameArg.getClass())) {
+      LogicalExpression swapArg = valueArg;
+      valueArg = nameArg;
+      nameArg = swapArg;
+      evaluator.functionName = COMPARE_FUNCTIONS_TRANSPOSE_MAP.get(functionName);
+    }
+    evaluator.success = nameArg.accept(evaluator, valueArg);
+    return evaluator;
+  }
+
+  public boolean isSuccess() {
+    // TODO Auto-generated method stub
+    return success;
+  }
+
+  public String getPath() {
+    return path;
+  }
+
+  public Long getValue() {
+    return value;
+  }
+
+  public String getFunctionName() {
+    return functionName;
+  }
+
+  @Override
+  public Boolean visitSchemaPath(SchemaPath path, LogicalExpression valueArg) throws RuntimeException {
+    this.path = path.getRootSegmentPath();
+
+    if(valueArg == null) {
+      return false;
+    }
+
+    switch (this.path) {
+      case "kafkaMsgOffset":
+        /*
+         * Do not pushdown "not_equal" on kafkaMsgOffset.
+         */
+        if(functionName.equals("not_equal")) {
+          return false;
+        }
+      case "kafkaPartitionId":
+        if(valueArg instanceof IntExpression) {
+          value = (long) ((IntExpression) valueArg).getInt();
+          return true;
+        }
+
+        if(valueArg instanceof LongExpression) {
+          value = ((LongExpression) valueArg).getLong();
+          return true;
+        }
+        break;
+      case "kafkaMsgTimestamp":
+        /*
+        Only pushdown "equal", "greater_than", "greater_than_or_equal" on kafkaMsgTimestamp
+         */
+        if(!functionName.equals("equal") && !functionName.equals("greater_than")
+               && !functionName.equals("greater_than_or_equal_to")) {
+          return false;
+        }
+
+        if(valueArg instanceof IntExpression) {
+          value = (long) ((IntExpression) valueArg).getInt();
+          return true;
+        }
```

Comment:
I think we no need to handle IntExpression, as timestamp cannot be Int.

Reply:
This was mostly added for unit tests. It is possible (although unlikely) for Kafka topic be configured for `CreateTime` and one can specify timestamp as Int.

I have, however, moved the check for `IntExpression` to the end since it is least probable.